### PR TITLE
Fix build under Linux/GCC Windows/MSVC

### PR DESCRIPTION
**Pull request by wilhuff**
All the recent churn around `Query` has broken the Linux/GCC and Windows/MSVC builds.

GCC and MSVC aren't happy about `using Query = testutil::Query;` when there's already `class Query` in scope.

Also fix a pile of other minor issues that have crept in.

### CODE REVIEW

**Comment by wilhuff** [comment]
Note the diffbase on #3499.

File: Firestore/core/src/firebase/firestore/objc/objc_class.h [hunk 1]
```diff
@@ -159,6 +164,7 @@ bool Equals(const Handle<T>& lhs, const Handle<T>& rhs) {
 #else  // !__clang__
 #define NS_ASSUME_NONNULL_BEGIN
 #define NS_ASSUME_NONNULL_END
+#define _Nullable
```
Comment (var-const) [inline Firestore/core/src/firebase/firestore/objc/objc_class.h]: Question: I remember we discussed putting the definition of `_Nullable` in a header or something like that, can you remind me what was the outcome of that?

File: cmake/compiler_setup.cmake [hunk 2]
```diff
@@ -116,9 +108,20 @@ if(MSVC)
 
     # unary minus operator applied to unsigned type, result still unsigned
     /wd4146
+
+    # character cannot be represented in the current code page
+    /wd4566
```
Comment (var-const) [inline cmake/compiler_setup.cmake]: Question: why is this change necessary?

File: Firestore/core/test/firebase/firestore/core/query_test.cc [hunk 3]
```diff
@@ -72,7 +71,7 @@ TEST(QueryTest, Constructor) {
 
 TEST(QueryTest, OrderBy) {
   auto query =
-      Query("rooms/Firestore/messages")
+      testutil::Query("rooms/Firestore/messages")
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/core/query_test.cc]: Re. the discussion on the name -- I'm sort of inclined to rename the helper method to avoid the issue. I don't feel strongly about this, though. Some suggestions: `MakeQuery`? `TestQuery`? `query`? (capitalization) `make_query`? `ParseQuery`? `Test<Query>`/`Make<Query>`/etc.?

File: cmake/compiler_setup.cmake (hunk 2 above)
Comment (wilhuff) [reply cmake/compiler_setup.cmake]: MSVC emits a warning that one of the unicode literals in one of our tests cannot be represented in the current code page (in my case Windows-1252). This is irrelevant to the test and is generally irrelevant, so I've suppressed the warning.

File: Firestore/core/test/firebase/firestore/core/query_test.cc (hunk 3 above)
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/core/query_test.cc]: I don't feel super strongly about it either, but I'm also not super excited about any of those options. It bugs me that it's an outlier among the rest of the methods, and it's only a few of the tests in `core` that actually see the conflict, but renaming the method affects all the tests.

File: Firestore/core/src/firebase/firestore/objc/objc_class.h (hunk 1 above)
Comment (var-const) [reply Firestore/core/src/firebase/firestore/objc/objc_class.h]: I'm fine with the current state, just checking.

File: Firestore/core/test/firebase/firestore/core/query_test.cc (hunk 3 above)
Comment (var-const) [reply Firestore/core/test/firebase/firestore/core/query_test.cc]: You could also create a local helper function in the affected tests that delegates to the troublesome `Query` and then use it within the file. It doesn't fully solve the consistency problem, though.

Barring that, I find qualifying the test function with the namespace to be a good approach.

Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/core/query_test.cc]: I'm going to stick with this for now then. Let's revisit if there are more outliers.
